### ISSUE TITLE
installation not allowed by "shutdown" plug rule of interface "shutdown" for "iot-agent" snap

### ISSUE THREAD

**Issue by couling** [CanonicalLtd/iot-agent#14]
Following instructions from IoT_DM_Whitepaper_11.05.20.pdf, installing on Ubuntu Core 22 (Intel)

I tried to
```shell
sudo snap install --edge --devmode iot-agent
```

But this resulted in an error
```
error: cannot perform the following tasks:
- Mount snap "iot-agent" (66) (installation not allowed by "shutdown" plug rule of interface "shutdown" for "iot-agent" snap)
```

Not using `--edge` release resulted in pulling an agent version dependant on `core20`, not `core22`.

Is this a known problem?

**Comment by bugraaydogar** [comment]
Hi Philip,

This is just a reference device management architecture and is not a very well-maintained solution. Therefore, we do not build it for different bases. Feel free to build your own version of iot-agent rather than relying on the snap available in the Global Store.

There was a recent PR regarding to the `shutdown` interface. https://github.com/snapcore/snapd/pull/12652
Basically, we will mark `shutdown` interface as a super-privileged interface. I assume, the store related changes are done and deployed whether we still require snapd related changes to land. 

In the meantime, you can remove `shutdown` interface from iot-agent since it is only used to reboot the device remotely for demonstration purpose. If you have your own IoT App Store, then feel free to create a ticket against Canonical, they will take care of it.

Thanks,
Bugra

**Comment by couling** [comment]
Apologies I really didn't understand that first paragraph:

> This is just a reference device management architecture and is not a very well-maintained solution. Therefore, we do not build it for different bases.

Are you saying the issue is with snapd itself, or this isn't the right place to report this issue?

**Comment by bugraaydogar** [comment]
Oh sorry, I had to be more specific. This is the correct place to report this problem. But, the whitepaper and all of the repositories are part of a reference solution that is not maintained. Therefore, even if you create issue reports here, they will only be fixed with our best effort without making any commitment. Ideally, you would need to fork these repositories and make the required adjustment yourself.

Besides that, I guess, we are discussing two different issues here;

- The current `iot-agent` snap is built against core20. This is why when you install it from the global store, the system will install core20 snap. That being said, it seems like one of our colleagues already adjusted the `snapcraft.yaml` file to use core22 as a base. However, a new build of the `iot-agent` snap is not uploaded to the store. This is why I emphasize that the device management solution is not well maintained. 

- The other topic is around `shutdown` interface. There is an open PR available here (https://github.com/snapcore/snapd/pull/12652) and when the changes land, the issue will disappear. If you have your own IoT App Store instance, you can communicate with Canonical's Store Team and they can resolve the issue related to `shutdown` interface. Currently, it is a known limitation. 

In summary, this is not a blocking issue if you are exploring the reference device management solution. Please build the `iot-agent` snap yourself and if necessary remove the `shutdown` interface from the `snapcraft.yaml` file.

**Comment by couling** [comment]
Thanks for that explanation. That's much appreciated.
